### PR TITLE
TapTimer

### PR DESCRIPTION
**Pull request by AhlaamK-tap**
added TapCountdowntimer in which user can set duration for timeout

### CODE REVIEW

File: app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java [hunk 1]
```diff
@@ -40,6 +42,8 @@ protected void onCreate(Bundle savedInstanceState) {
         expirationDate = findViewById(R.id.expiration_date);
         cardLayout = findViewById(R.id.card_Layout);
         textRecognitionML = new TapTextRecognitionML(this);
+
```
Comment (jimmy-tap) [inline app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java]: Why to add two empty lines?

File: cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     *                          {@link #onTick(long)} callbacks.
+     */
+    public TapCountDownTimer(long millisInFuture, long countDownInterval, Activity context) {
+        super(millisInFuture, countDownInterval);
+        this.context = context;
+    }
+
+    @Override
+    public void onTick(long millisUntilFinished) {
+        Toast.makeText(context, "Seconds remaining:"+millisUntilFinished / 1000, Toast.LENGTH_SHORT).show();
```
Comment (jimmy-tap) [inline cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java]: No need for toast on each tick

File: cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Override
+    public void onFinish() {
+        Toast.makeText(context, "Timed out", Toast.LENGTH_SHORT).show();
+        context.setResult(Activity.RESULT_CANCELED);
+        context.finish();
```
Comment (jimmy-tap) [inline cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java]: After timeout we need to call removeInlineScanner(), why you are closing the main activity?

**Review by jimmy-tap — CHANGES_REQUESTED** [review]
Thanks for your effort, there is a few changes required

File: app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java (hunk 1 above)
Comment (AhlaamK-tap) [reply app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java]: discarded lines

File: cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java (hunk 2 above)
Comment (AhlaamK-tap) [reply cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java]: ok removed

File: cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java (hunk 3 above)
Comment (AhlaamK-tap) [reply cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java]: resolved

File: cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import java.net.URISyntaxException;
+
+
+/**
+ * Created by AhlaamK on 4/2/20.
+ * <p>
+ * Copyright (c) 2020    Tap Payments.
+ * All rights reserved.
+ **/
+public class TapCountDownTimer extends CountDownTimer {
```
Comment (jimmy-tap) [inline cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java]: Why we have TapCountDownTimer, it's useless now

File: app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java [hunk 5]
```diff
@@ -88,7 +91,9 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
                 Bitmap bitmap = ImagePicker.getImageFromResult(this, resultCode, data);
                 textRecognitionML.decodeImage(bitmap);
                 break;
-
+            case Activity.RESULT_CANCELED:
```
Comment (jimmy-tap) [inline app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java]: Who is using this case?

**Review by jimmy-tap — CHANGES_REQUESTED** [review]
Please review the comments

File: app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java (hunk 5 above)
Comment (AhlaamK-tap) [reply app/src/main/java/company/tap/tapcardscannerkit_android/MainActivity.java]: forgot to remove..edited now

File: cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java (hunk 4 above)
Comment (AhlaamK-tap) [reply cardscanner/src/main/java/company/tap/cardscanner/TapCountDownTimer.java]: resolved

**Pull request merged**